EBLINUX-27055: adding credential support (#90)
Elektrobit/ebcl_template: docs/images/params.md

```
In addition, an armored public key file or URL can be given as “key”, and a unarmored gpg file can be given as “gpg”, to authenticate the package sources.

> NOTE:
> 
> If apt repositories require credentials, please store the [credential files](https://manpages.debian.org/testing/apt/apt_auth.conf.5.de.html) (\*.conf files in netrc format) under ~/.ebcl_config/auth.d/ (or) directly create/copy files under the /workspace/tools/user_config/auth.d directory of devcontainer.
> 
> Example: when use apt repos from artifactory-central.elektrobit.com (protected by credentials) as apt_repo in any of the config file.
> 
> a) Generate access token by clicking *Generate an Identity token* button in artifactory's [user_profile](https://artifactory-central.elektrobit.com/ui/user_profile)
> 
> b) Add a credential file called artifactory.conf like below under ~/.ebcl_config/auth.d/ of host machine (or) /workspace/tools/user_config/auth.d of devcontainer before starting the image creation operation

Reviewer: ```suggestion
NOTE:

If apt repositories require credentials store [credential files](https://manpages.ubuntu.com/manpages/jammy/man5/netrc-ssl.5.html) (\*.conf files in netrc format) can be placed in ~/.ebcl_config/auth.d/ on the host before the container is started or directly created in the /workspace/tools/user_config/auth.d directory of the devcontainer.

Example when using apt repositories from artifactory-central.elektrobit.com:

a) Generate access token by clicking *Generate an Identity token* button in artifactory's [user_profile](https://artifactory-central.elektrobit.com/ui/user_profile)

b) Add a credential file called artifactory.conf in ~/.ebcl_config/auth.d/ of the host or in /workspace/tools/user_config/auth.d of the devcontainer
```

Some minor wording changes and against my previous comment change link to the manpage of netrc.
Reason for that: netrc is more limited than apt auth.conf, for example it allows only the host name, not with protocol (e.g. https), port or path. The file is interpreted as pure netrc in twolocations: 1. debootstrap uses wget and 2. python-requests for all pacakges downloaded by the python tools

Reviewer 2: Can not agree with those wording changes. Some minor details:

```suggestion
NOTE:

If apt repositories require credentials [credential files](https://manpages.ubuntu.com/manpages/jammy/man5/netrc-ssl.5.html), (\*.conf files in netrc format) can be placed in ~/.ebcl_config/auth.d/ on the host before the container is started or directly created in the /workspace/tools/user_config/auth.d directory of the devcontainer.

Example when using apt repositories from artifactory-central.elektrobit.com:

a) Generate an access token by clicking the *Generate an Identity token* button in artifactory's [user_profile](https://artifactory-central.elektrobit.com/ui/user_profile)

b) Add a credential file called artifactory.conf in ~/.ebcl_config/auth.d/ on the host or in /workspace/tools/user_config/auth.d of the devcontainer
```

Reviewer: I know Vignesh already complained about my broken first sentence. But yours doesn't really make sense as well.
For the first sentence I'd go with:

```
If apt repositories require authentication, [credential files](https://manpages.ubuntu.com/manpages/jammy/man5/netrc-ssl.5.html) (\*.conf files in netrc format) can be placed in ~/.ebcl_config/auth.d/ on the host before the container is started or directly created in the /workspace/tools/user_config/auth.d directory of the devcontainer.
```

Without markup:
 > If apt repositories require authentication, credential files (*.conf files in netrc format) can be placed in ~/.ebcl_config/auth.d/ on the host before the container is started or directly created in the /workspace/tools/user_config/auth.d directory of the devcontainer.

I replaced credentials with authentication, to not have "credentials" and "credential right after each other and moved the comma. I' not sure though if the comma is required at all, but for sure it is not required after "createnial files" like in your comment

Reviewer 2: Yes, the comma from me is for sure wrong, I am fine with above first sentence.

```
In addition, an armored public key file or URL can be given as “key”, and a unarmored gpg file can be given as “gpg”, to authenticate the package sources.

NOTE: In case any of the apt repositories require credentials, please keep them in the ~/.ebcl_config/auth.conf file (in .netrc format)

> 

Reviewer: This needs more details and the main use case here will be artifactory, so at least some instructions how to use this for artifactory should be added.
I wouldn't go for a step-by-step manual, but something like: Get an auth token from artifactory and a brief example how the file looks like. For details this should link to https://manpages.debian.org/testing/apt/apt_auth.conf.5.en.html

Author: accepted and same is updated
